tensorflow/lite/micro/micro_allocator.cc

```diff
@@ -510,13 +510,14 @@ TfLiteStatus MicroAllocator::FinishModelAllocation(
   }
 
   // Allocate scratch buffer metadata and buffers for variable tensors.
+  TF_LITE_ENSURE_STATUS(AllocateScratchBufferHandles(
+      scratch_buffer_handles, scratch_buffer_request_count_));
+
```

Comment:
`scratch_buffer_request_count_` is the number of scratch buffers for all subgraphs together, so it should not be inside the subgraph loop.